components/camel-dynamic-router/src/main/docs/dynamic-router-control-component.adoc

```diff
@@ -0,0 +1,247 @@
+// component options: END
+
+// endpoint options: START
+// endpoint options: END
+
+=== Subscribing
+
+Subscribing can be achieved by using query parameters in the control endpoint URI, or by sending a
+`DynamicRouterControlMessage` to the control endpoint URI.
+
+===== URI examples
+
+.Example Java URI `RouteBuilder` Subscription
+[source,java]
+----
+// Send a subscribe request to the dynamic router that will match every exchange and route messages to the URI: "direct:myDestination"
+from("direct:subscribe").to("dynamic-router-control:subscribe?subscribeChannel=myChannel&subscriptionId=mySubId&destinationUri=direct:myDestination&priority=5&predicate=true&expressionLanguage=simple");
+----
+
+.Example Java URI `ProducerTemplate` Subscription
+[source,java]
+----
+CamelContext context = new DefaultCamelContext();
+context.start();
+ProducerTemplate template = context.createProducerTemplate();
+RouteBuilder.addRoutes(context, rb -> {
+    // Route for subscriber destination
+    rb.from("direct:myDestination")
+            .to("log:dynamicRouterExample?showAll=true&multiline=true");
+    // Route for subscribing
+    rb.from("direct://subscribe")
+            .toD("dynamic-router-control://subscribe" +
+                    "?subscribeChannel=${header.subscribeChannel}" +
+                    "&subscriptionId=${header.subscriptionId}" +
+                    "&destinationUri=${header.destinationUri}" +
+                    "&priority=${header.priority}" +
+                    "&predicateBean=${header.predicateBean}");
+});
+Predicate predicate = PredicateBuilder.constant(true);
+context.getRegistry().bind("predicate", predicate);
+template.sendBodyAndHeaders("direct:subscribe", "",
+        Map.of("subscribeChannel", "test",
+                "subscriptionId", "testSubscription1",
+                "destinationUri", "direct:myDestination",
+                "priority", "1",
+                "predicateBean", "predicate"));
+----
+Above, because the control URI is dynamic, and since a `ProducerTemplate` does not have a built-in way to send to a
+dynamic URI, we have to send subscription parameters from a `ProducerTemplate` in a different way.  The dynamic-aware
+endpoint uses headers "under the hood", because the URI params are converted to headers, so we can set the headers
+deliberately.
+
+.Example XML URI Subscription
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+		http://camel.apache.org/schema/spring http://camel.apache.org/schema/spring/camel-spring.xsd
+		http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd ">
+
+    <camelContext id="camelContext" xmlns="http://camel.apache.org/schema/spring">
+        <route>
+            <from uri="direct:subscribe"/>
+            <toD uri="dynamic-router-control:subscribe
```

Comment:
There seems to not be any dynamic in this, so it can maybe be a static to

Reply:
Did you see the whole URI?
```
            <toD uri="dynamic-router-control:subscribe
                      ?subscribeChannel=myChannel
                      &amp;subscriptionId=mySubId
                      &amp;destinationUri=direct:myDestination
                      &amp;priority=5
                      &amp;predicate=true
                      &amp;expressionLanguage=simple"/>
```
That's definitely dynamic, so I need it there for the documentation to be correct.

Reply:
The dynamic aware part takes a look at the query parameters and removes them while adding them to headers and the query parameters that you see here are the ones that it uses when it's converting to the plain endpoint without query parameters.